### DISCUSSION TITLE
`Surface::get_current_texture` is really slow

### DISCUSSION THREAD

**Discussion by botahamec** [gfx-rs/wgpu#3130] [Q&A] [answered]
I made a bunnymark program last night. It seems like a lot of my performance is going to the `get_next_texture` function, which is ran by `surface.get_current_texture`. This is with Mailbox mode on a Windows 10 desktop with a GTX 1650 running on Vulkan. It starts out normal, taking only several microseconds, but over time, increases to as much as 40 milliseconds. Any ideas as to what could cause this?

![Tracy_aMllEI3n0s](https://user-images.githubusercontent.com/32026883/197276595-fb728bcf-51c8-4e3e-aed3-b7901edb41be.png)

**Comment by cwfitzgerald** [comment, marked as answer]
This is basically because you're gpu bound. When the gpu gets too far behind the cpu, the driver will start "holding the cpu back" by means of get_next_texture.

**Reply by botahamec** [reply]
Hmm. I assumed that was done on `Queue::submit` or `SurfaceTexture::present`. That certainly explains it. Thanks.
